Fixify Self Contained

Siliconrob opened the pull request:
The reason the self contained artifact was behaving differently than the full publish was that the config file which pulls the log in was not packaged in the distributable and since the distributable is really a zip file it needs this file within the contents by default unless you have it search out some paths.

Anyway this is quick fix and is going into my lightning talk as a gotcha

![](https://media.giphy.com/media/3ohhwks5vrYYXUuFlm/giphy.gif)



Also I added a lot of Rider ignores, but I can trim them out if you would like.

Siliconrob committed:
Fixify Self Contained

Siliconrob committed:
Ignore Rider folder and adjust build for single file exe

Siliconrob committed:
Update .gitignore

Siliconrob committed:
Bye bye .idea files

Siliconrob commented:
💥 Fixed my dumb mistakes 💥 

👓

Dessyreqt commented on psakefile.ps1:
```diff
@@ -4,10 +4,10 @@ properties {
     $project_dir = "$solution_dir\$solution_name"
     
-    $output_dir = "$project_dir\bin\Debug\netcoreapp3.1"
+    $output_dir = "$project_dir\bin\Debug\netcoreapp3.1\win-x64\publish"
```
This breaks the default flow, which just does a clean and build, not a publish, so if you haven't published before it won't be able to execute the last step which is to build the sample website.

You can test this yourself by deleting your build folder, opening a command line at the repository root and running `build.bat`.

Siliconrob replied:
Removed the `publish` portion of the path and tested from clean, build, and then publish

Siliconrob committed:
Update psakefile.ps1

Removed the publish portion to keep in default build behavior along with keeping publish
